projects/fsastorefrontlib/src/cms-components/my-account/inbox/unread-messages-indicator/unread-messages-indicator.component.ts

```diff
@@ -0,0 +1,50 @@
+import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
+import { AuthService, UserService } from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { InboxService } from '../../../../core/my-account/facade/inbox.service';
+
+@Component({
+  selector: 'cx-fs-unread-messages-indicator',
+  templateUrl: './unread-messages-indicator.component.html',
+})
+export class UnreadMessagesIndicatorComponent implements OnInit, OnDestroy {
+  numberOfUnreadMessages$: Observable<number>;
+  user$: Observable<boolean>;
+  subscription = new Subscription();
+
+  constructor(
+    protected inboxService: InboxService,
+    protected changeDetectorRef: ChangeDetectorRef,
+    protected authService: AuthService,
+    protected userService: UserService
+  ) {}
+
+  ngOnInit(): void {
+    this.user$ = this.authService.isUserLoggedIn();
+    this.subscription.add(
+      this.inboxService.unreadMessagesState.subscribe(isMessageRead => {
```

Comment:
Again, it's better to use pipe which will give us the possibility to chain different operators than to use subscribe which only listen emitted values.